### PR TITLE
refactor(pkg_aliases): create a macro for creating whl aliases

### PR DESCRIPTION
**Pull request by aignas**
This just cleans up the code and moves more logic from the repository_rule
(i.e. generation of `BUILD.bazel` files) to loading time (macro evaluation).
This makes the unit testing easier and I plan to also move the code that is
generating config setting names from filenames to this new macro, but wanted to
submit this PR to reduce the review chunks.

Summary:
- Add a new `pkg_aliases` macro.
- Move logic and tests for creating WORKSPACE aliases.
- Move logic and tests bzlmod aliases.
- Move logic and tests bzlmod aliases with groups.
- Add a test for extra alias creation.
- Use `whl_alias` in `pypi` extension integration tests.
- Improve the serialization of `whl_alias` for passing to the pypi hub repo.

Related to #260, #2386, #2337, #2319 - hopefully cleaning the code up will make
it easier to address those feature requests later.

### CODE REVIEW

**Comment by aignas** [comment]
Bazel 9 integration tests are failing - I'll wait for the #2379.

File: python/private/pypi/pkg_aliases.bzl [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        actual,
+        group_name = None,
+        extra_aliases = None,
+        native = native,
+        select = select):
+    """Create aliases for an actual package.
+
+    Args:
+        name: {type}`str` The name of the package.
+        actual: {type}`dict[Label, str] | str` The config settings for the package
```
Comment (rickeylev) [inline python/private/pypi/pkg_aliases.bzl:73]:
```suggestion
        actual: {type}`dict[Label, str] | str` The name of the repo the aliases point to, or a dict of select conditions to repo names for the aliases to point to
```

File: python/private/pypi/pkg_aliases.bzl [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                    config_setting: "@{repo}//:{target_name}".format(
+                        repo = repo,
+                        target_name = name,
+                    )
+                    for config_setting, repo in actual.items()
+                },
+                no_match_error = no_match_error,
+            )
+        else:
+            fail("BUG: should have a dictionary or a string")
```
Comment (rickeylev) [inline python/private/pypi/pkg_aliases.bzl:115]:
```suggestion
            fail("The `actual` arg must be a dictionary or a string")
```

File: tests/pypi/pkg_aliases/pkg_aliases_test.bzl [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+_tests.append(_test_legacy_aliases)
+
+def _test_config_setting_aliases(env):
+    # Use this function as it is used in pip_repository
+    actual = []
+    actual_no_match_error = []
+
+    def mock_select(value, no_match_error = None):
+        actual_no_match_error.append(no_match_error)
+        env.expect.that_str(no_match_error).equals("""\
```
Comment (rickeylev) [inline tests/pypi/pkg_aliases/pkg_aliases_test.bzl:75]: Oh ouch, this is brittle. I gotta add some StrSubject methods to make partial matching easier. FYI: .contains() can be used for less exact matching

**Review by rickeylev — APPROVED** [review]

**Pull request merged**